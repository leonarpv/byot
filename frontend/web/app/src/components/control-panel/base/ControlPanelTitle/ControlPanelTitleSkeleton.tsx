import * as React from 'react';
import {Box, makeStyles, Theme} from '@material-ui/core';
import {WithStyles} from '@byot-frontend/web-common/src/types/WithStyles';
import {PatchSkeleton} from '@byot-frontend/web-common/src/components/elementary/patch/PatchSkeleton';

interface Props extends WithStyles<typeof styles> {}

const styles = (theme: Theme) => ({
  skeleton: {
    width: '100%',
    height: theme.spacing(8),
  },
});
const useStyles = makeStyles(styles);

export const ControlPanelTitleSkeleton: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  return (
    <Box mb={4} mt={2}>
      <PatchSkeleton variant="rect" classes={{root: styles.skeleton}} />
    </Box>
  );
};
