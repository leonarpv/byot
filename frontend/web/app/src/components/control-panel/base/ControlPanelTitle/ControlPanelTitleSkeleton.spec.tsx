import * as React from 'react';
import {render} from '@testing-library/react';
import {ControlPanelTitleSkeleton} from './ControlPanelTitleSkeleton';

describe('<ControlPanelTitleSkeleton/>', () => {
  it('should render', () => {
    const {container} = render(<ControlPanelTitleSkeleton />);
    expect(container).toMatchSnapshot();
  });
});
