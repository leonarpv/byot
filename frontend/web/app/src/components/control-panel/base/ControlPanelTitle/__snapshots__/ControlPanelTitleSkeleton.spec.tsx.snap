// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlPanelTitleSkeleton/> should render 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <span
      class="MuiSkeleton-root makeStyles-root-3 makeStyles-skeleton-1 MuiSkeleton-rect MuiSkeleton-pulse"
    />
  </div>
</div>
`;
